### ISSUE TITLE
Unused parameters affecting the forward pass

### ISSUE THREAD

**Issue by ParitoshParmar** [pytorch/pytorch#9976]
I am using only a part of a pretrained net. If I keep the unused parameters, then I am getting the following results:
Output of method 1:
Epoch: 0 Iteration: 0 Loss: tensor(18.7229, device=‘cuda:0’)
However, if I get rid of the unused parameters, then I am getting the following results:
Output of method 2:
Epoch: 0 Iteration: 0 Loss: tensor(30.2404, device=‘cuda:0’)

Why would the unused parameters affect the outcome of the forward pass?

**Comment by zou3519** [comment]
Could you please provide a reproducible example? Unused parameters shouldn't affect the outcome of the forward pass.

**Comment by ParitoshParmar** [comment]
@zou3519 , I made a minimal reproducible example. But I am not getting that much difference in the minimal example, while I am still getting the difference in my main code. For data to be used in the minimal example, I am just saving the tensor from my main code and loading it in the minimal example. So the only difference seems to be in the data loading. I haven't figured out if and where I am making mistake in my main code. So, I am closing this. Thank you very much for the support.
